internal/events/nats/natschannel_test.go

```diff
@@ -89,22 +89,32 @@ func TestNatsChannel(t *testing.T) {
 	//
 	// Note that message delivery order is not important (and may not be deterministic /
 	// meaningful in a multi-process world).
-	results := make([]*message.Message, 0, 3)
-	for i := 0; i < 3; i++ {
+	results := make([]*message.Message, 0, 7)
+loop:
+	for i := 0; i < 7; i++ {
 		select {
 		case m := <-out1:
 			results = append(results, m)
 			t.Logf("Got %s from out1", m.Payload)
 		case m := <-out2:
 			results = append(results, m)
 			t.Logf("Got %s from out2", m.Payload)
-		case <-time.After(20 * time.Second):
-			t.Fatalf("timeout waiting for message %d", i)
+		case <-time.After(25 * time.Second):
+			t.Logf("timeout waiting for message %d", i)
+			break loop
```

Comment:
Using a `Fatal` here means that if we don't get redeliveries, we may error out on the tests.  I'm not sure why I didn't see this locally, but _did_ see this on CI.  CI is correct.